sway-core/tests/sway_to_ir/mutable_struct.ir

```diff
@@ -27,4 +27,4 @@ script {
 !5 = span !0 98 112
 !6 = span !0 109 111
 !7 = span !0 117 123
-!8 = span !0 124 125
+!8 = span !0 161 162
```

Comment:
It looks like the span for reading a struct field was way off.  Weird.  I saw above in `optimize.rs` some changes related to the owned struct/enum stuff, so that must be it.

Reply:
I removed the owned stuff since it was a holdover from when we had lifetimes associated with spans...those owned fields previously had no span data at all, so it makes sense they were off